src/main/groovy/edu/oregonstate/mist/api/Application.groovy

```diff
@@ -20,67 +19,58 @@ import io.dropwizard.setup.Environment
 import javax.ws.rs.WebApplicationException
 
 /**
- * Main application class.
+ * Main application base class.
  */
-class SkeletonApplication extends Application<Configuration> {
+class Application<T extends Configuration> extends io.dropwizard.Application<T> {
     /**
      * Initializes application bootstrap.
      *
      * @param bootstrap
      */
     @Override
-    public void initialize(Bootstrap<Configuration> bootstrap) {
+    public void initialize(Bootstrap<T> bootstrap) {
         bootstrap.addBundle(new TemplateConfigBundle())
     }
 
     /**
-     * Registers lifecycle managers and Jersey exception mappers
-     * and container response filters
+     * Registers basic API stuff like the InfoResource, build info lifecycle
+     * manager, Jersey exception mappers, pretty print filter,
+     * and authentication handler
      *
+     * @param configuration
      * @param environment
-     * @param buildInfoManager
      */
-    protected void registerAppManagerLogic(Environment environment,
-                                           BuildInfoManager buildInfoManager) {
+    protected void setup(T configuration, Environment environment) {
```

Comment:
This is a good method / refactoring, but we should do this at the skeleton level so that applications can benefit from it.